src/OrleansAzureUtils/Providers/Streams/AzureQueue/AzureQueueBatchContainer.cs

```diff
@@ -102,6 +112,14 @@ public bool ImportRequestContext()
             return false;
         }
 
+        public void GetObjectData(SerializationInfo info, StreamingContext context)
```

Comment:
But that can change in the future. 
The class should be correct regardless of how (maybe incorrectly) it is used. The pattern that is used on this line is wrong - if this field is only set once at construction, this should be enforced via `readonly`. Otherwise, it can be set anytime (even if we don't do it now) and then the class serialization should be correct.